plugin.video.BlanketFort/plugin.video.BlanketFort/default.py

```diff
@@ -0,0 +1,454 @@
+        liz.setProperty( "Fanart_Image", fanart )
+        liz.setProperty('IsPlayable', 'true')
+
+
+
+  # Thanks to daschacka, an epg scraper for http://i.teleboy.ch/programm/station_select.php - http://forum.xbmc.org/showpost.php?p=936228&postcount=1076
+def getepg(link):
+        url=urllib.urlopen(link)
+        source=url.read()
+        url.close()
+        source2 = source.split("Jetzt")
+        source3 = source2[1].split('programm/detail.php?const_id=')
+        sourceuhrzeit = source3[1].split('<br /><a href="/')
+        nowtime = sourceuhrzeit[0][40:len(sourceuhrzeit[0])]
+        sourcetitle = source3[2].split("</a></p></div>")
+        nowtitle = sourcetitle[0][17:len(sourcetitle[0])]
+        nowtitle = nowtitle.replace("ö","oe")
+        nowtitle = nowtitle.replace("ä","ae")
+        nowtitle = nowtitle.replace("ü","ue")
+        return "  - "+nowtitle+" - "+nowtime
+
+
+xbmcplugin.setContent(int(sys.argv[1]), 'movies')
+try:
+    xbmcplugin.addSortMethod(int(sys.argv[1]), xbmcplugin.SORT_METHOD_UNSORTED)
+except:
+    pass
+try:
+    xbmcplugin.addSortMethod(int(sys.argv[1]), xbmcplugin.SORT_METHOD_LABEL)
+except:
+    pass
+try:
+    xbmcplugin.addSortMethod(int(sys.argv[1]), xbmcplugin.SORT_METHOD_DATE)
+except:
+    pass
+try:
+    xbmcplugin.addSortMethod(int(sys.argv[1]), xbmcplugin.SORT_METHOD_GENRE)
+except:
+    pass
+
+params=get_params()
+
+url=None
+name=None
+mode=None
+playlist=None
+
+try:
+    url=urllib.unquote_plus(params["url"]).decode('utf-8')
+except:
+    pass
+try:
+    name=urllib.unquote_plus(params["name"])
+except:
+    pass
+try:
+    iconimage=urllib.unquote_plus(params["iconimage"])
+except:
+    pass
+try:
+    fanart=urllib.unquote_plus(params["fanart"])
+except:
+    pass
+try:
+    mode=int(params["mode"])
+except:
+    pass
+try:
+    playlist=eval(urllib.unquote_plus(params["playlist"]).replace('|',','))
+except:
+    pass
+
+xbmc.log("Mode: "+str(mode))
+if not url is None:
+    xbmc.log("URL: "+str(url.encode('utf-8')))
+xbmc.log("Name: "+str(name))
+
+
+if mode==None:
+    xbmc.log("getData")
+    data = 'http://pastebin.com/raw/Cnm4FeBJ' 
+    getData(data, fanart)
+
+elif mode==2:
+    xbmc.log("getChannelItems")
+    getChannelItems(name,url,fanart)
+
+elif mode==3:
+    xbmc.log("")
+    getSubChannelItems(name,url,fanart)
+
+elif mode==4:
+    xbmc.log("")
+    getFavorites()
+
+elif mode==5:
+    xbmc.log("")
+    try:
+        name = name.split('\\ ')[1]
+    except:
+        pass
+    try:
+        name = name.split('  - ')[0]
+    except:
+        pass
+    addFavorite(name,url,iconimage,fanart)
+
+elif mode==6:
+    xbmc.log("rmFavorite")
+    try:
+        name = name.split('\\ ')[1]
+    except:
+        pass
+    try:
+        name = name.split('  - ')[0]
+    except:
+        pass
+    rmFavorite(name)
+
+elif mode==7:
+    xbmc.log("addSource")
+    addSource(url)
+
+elif mode==8:
+    xbmc.log("rmSource")
+    rmSource(name)
+
+elif mode==9:

```

Comment:
I don't think these modes are used and should be removed. I don't see the supporting routines anywhere.